tests/test_user_delete.py

```diff
@@ -0,0 +1,206 @@
+
+
+def test_del_request_others_researcher(client):
+    """Unit admin tries to delete research user"""
+
+    email_to_delete = "researchuser@mailtrap.io"
+
+    with pytest.raises(ddserr.UserDeletionError):
+        response = client.post(
+            tests.DDSEndpoint.USER_DELETE,
+            headers=tests.UserAuth(tests.USER_CREDENTIALS["delete_me_unitadmin"]).token(client),
+            data=json.dumps({"email": email_to_delete}),
+            content_type="application/json",
+        )
+    assert response.status_code == http.HTTPStatus.BAD_REQUEST
+
+    # verify that user was not deleted
+    exists = utils.email_return_user(email_to_delete)
+    assert exists is not None
+    assert type(exists).__name__ == "ResearchUser"
+    assert exists.primary_email == email_to_delete
+
+
+def test_del_request_others_researcher(client):
+    """Unit admin tries to delete unit user from different unit"""
+
+    email_to_delete = "unituser1@mailtrap.io"
+
+    # with pytest.raises(ddserr.UserDeletionError):
+    response = client.post(
+        tests.DDSEndpoint.USER_DELETE,
+        headers=tests.UserAuth(tests.USER_CREDENTIALS["delete_me_unitadmin"]).token(client),
+        data=json.dumps({"email": email_to_delete}),
+        content_type="application/json",
+    )
+    print(response.data)
+    assert response.status_code == http.HTTPStatus.BAD_REQUEST
+
+    # verify that user was not deleted
+    exists = utils.email_return_user(email_to_delete)
+    assert exists is not None
+    assert type(exists).__name__ == "UnitUser"
+    assert exists.primary_email == email_to_delete
+
+
+def test_del_request_others_self(client):
+    """Unit admin tries to instantly self-delete via this endpoint"""
+
+    email_to_delete = "delete_me_unitadmin@mailtrap.io"
+
+    # with pytest.raises(ddserr.UserDeletionError):
+    response = client.post(
+        tests.DDSEndpoint.USER_DELETE,
+        headers=tests.UserAuth(tests.USER_CREDENTIALS["delete_me_unitadmin"]).token(client),
+        data=json.dumps({"email": email_to_delete}),
+        content_type="application/json",
+    )
+    assert response.status_code == http.HTTPStatus.BAD_REQUEST
+
+    # verify that user was not deleted
+    exists = utils.email_return_user(email_to_delete)
+    assert exists is not None
+    assert type(exists).__name__ == "UnitUser"
+    assert exists.primary_email == email_to_delete
+
+
+def test_del_request_others_success(client):
+    """Unit admin deletes unit user"""
+
+    email_to_delete = "delete_me_unituser@mailtrap.io"
+
+    response = client.post(
+        tests.DDSEndpoint.USER_DELETE,
+        headers=tests.UserAuth(tests.USER_CREDENTIALS["delete_me_unitadmin"]).token(client),
+        data=json.dumps({"email": email_to_delete}),
+        content_type="application/json",
+    )
+    assert response.status_code == http.HTTPStatus.OK
+
+    # Make sure that user was deleted
+    exists = utils.email_return_user(email_to_delete)
+    assert exists is None
```

Comment:
Would be nice to add the case we struggled with here, namely that the user is deleted but not the email related to that user.